### PR TITLE
Refactor EncryptionProfile entity

### PR DESCRIPTION
**Pull request by svendecabooter**
This pull request refactors the EncryptionProfile interface & EncryptionMethod plugin, by changing the following:

* Let the functions getEncryptionMethod() & getEncryptionKey() on EncryptionProfile entity return a reference to the actual plugin or entity respectively, instead of just the ID. This avoids various scattered pieces of code trying to interact with these objects to have to load them based on the given ID, and just interact with the returned object instead.
* The ID properties are still what's being stored in the config entity, and they are still publicly available via the getEncryptionMethodId() and getEncryptionKeyId() methods.
* The EncryptionMethodInterface has an added method getLabel(), to be able to present a human-readable string for a given encryption method (as defined in the annotation).
* An abstract base class EncryptionMethodBase has been added, so other EncryptionMethod plugins can extend that one, and already get a default implementation of getLabel(). This can also be useful in the future, when new functionality needs to be introduced for all EncryptionMethod plugins, without having to ask the author of each plugin to implement the changes.

These changes affect the following:
* The EncryptService service could now be simplified, since the EncryptionProfile can supply the encryption method & key objects directly.
* The EncryptionProfileListBuilder (rendering the Encryption Profiles overview page) can now list the used encryption profile & key in a human-readable way, instead of using the internal machine name.
* Tests were cleaned up and simplified to reflect these changes.

Also in this PR:
* General code cleanup
* Fix for wrongly assigned key default value in form (see https://github.com/d8-contrib-modules/encrypt/pull/91)
* Fix for typos in unit tests (see https://github.com/d8-contrib-modules/encrypt/pull/90)

### CODE REVIEW

File: src/Plugin/EncryptionMethod/McryptAES256Encryption.php [hunk 1]
```diff
@@ -3,7 +3,6 @@
 namespace Drupal\encrypt\Plugin\EncryptionMethod;
 
 use Drupal\encrypt\EncryptionMethodInterface;
-use Drupal\Core\Plugin\PluginBase;
```
Comment (nerdstein) [inline src/Plugin/EncryptionMethod/McryptAES256Encryption.php]: Can we delete this file? I think we are moving away from mcrypt altogether

**Comment by nerdstein** [comment]
This looks really good. Can you remove the mcrypt method? That will knock out another issue while we're at it

**Comment by nerdstein** [comment]
@svendecabooter where is composer getting invoked in travis? we should use this strategy: https://github.com/composer/composer/issues/1314

**Comment by svendecabooter** [comment]
I removed the Mcrypt EncryptionMethod as per your request.
We should work on integrating the Defuse / OpenSSL encryption method as a contrib module, so we have some real-life implementations of the APIs provided by the Encrypt module.
I have some basic WIP code for that... let me know if you want to get involved with creating that module.

Reading up on the composer / Travis thing now.
Haven't used Travis before, so I'll see if I can make sense of it.

**Comment by svendecabooter** [comment]
Trying to fix the token issue as described in this link: https://blog.wyrihaximus.net/2015/09/github-auth-token-on-travis/

However, I'm stuck at "Add token to Travis", since I don't seem to have permission to do that...

**Comment by nerdstein** [comment]
Yes, i'd be interested in contributing to that module. Do you want me to set up a new d.o project?

**Comment by svendecabooter** [comment]
@nerdstein sure go ahead... if you add me as contributor, i'll push the first draft code i've been working on.

**Comment by nerdstein** [comment]
https://www.drupal.org/sandbox/nerdstein/2669882

**Comment by svendecabooter** [comment]
Yay build now passes!
Trying to activate phpunit as well now...

Thanks for merging this PR if you don't see any more problems with it.

**Comment by nerdstein** [comment]
merged, thanks!! looking good

**Pull request merged**